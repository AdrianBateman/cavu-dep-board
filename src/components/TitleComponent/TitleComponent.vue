<template>
  <header v-if="title"
          class="title-component">
    <img v-if="icon && icon.src"
         class="title-component__icon"
         :src="icon.src"
         :width="icon.width"
         :height="icon.height"
         :alt="icon.alt" />
    <h1 class="title-component__text">{{title}}</h1>
  </header>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      required: true
    },
    icon: {
      type: Object,
      default: () => ({})
    }
  }
}
</script>

<style scoped lang="scss">
$desktopWidth: 500px;

.title-component {
  display: flex;
  align-items: center;
  padding: 32px 0 30px 45px;
  background: radial-gradient(hsl(52, 79%, 62%) 0%, hsl(51, 73%, 53%) 100%);

  @media screen and (min-width: $desktopWidth) {
    padding: 42px 0 30px 55px;
  }

  &__icon {
    width: 100%;
    max-width: 2rem;
    height: auto;

    @media screen and (min-width: $desktopWidth) {
      max-width: 3rem;
    }
  }

  &__text {
    margin: 0 0 0 40px;
    font-size: 2rem;

    @media screen and (min-width: $desktopWidth) {
      font-size: 3rem;
      margin: 0 0 0 30px;
    }
  }
}
</style>
