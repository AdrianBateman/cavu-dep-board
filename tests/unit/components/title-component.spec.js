import { shallowMount } from '@vue/test-utils'
import TitleComponent from '../../../src/components/TitleComponent/TitleComponent.vue'

const title = 'Foo bar'
const icon = {
  src: 'https://via.placeholder.com/40x40',
  width: 40,
  height: 40,
  alt: 'test image'
}

describe('TitleComponent', () => {
  it('renders with supplied title prop', () => {
    const wrapper = shallowMount(TitleComponent, {
      propsData: {
        title
      }
    });

    const h1Tag = wrapper.find('h1');

    expect(h1Tag.text())
      .toMatch(title);
  })

  it('will not render when missing required title prop', () => {
      const wrapper = shallowMount(TitleComponent);

      expect(wrapper)
          .toMatchSnapshot();
  });

  it('renders with supplied icon and title props', () => {
      const wrapper = shallowMount(TitleComponent, {
          propsData: {
              title,
              icon,
          },
      });

      expect(wrapper)
        .toMatchSnapshot();
  });
})
