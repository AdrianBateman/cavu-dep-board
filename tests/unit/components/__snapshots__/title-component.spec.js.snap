// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleComponent renders with supplied icon and title props 1`] = `
<header class="title-component"><img src="https://via.placeholder.com/40x40" width="40" height="40" alt="test image" class="title-component__icon">
  <h1 class="title-component__text">Foo bar</h1>
</header>
`;

exports[`TitleComponent will not render when missing required title prop 1`] = ``;
